src/lightning_app/runners/cloud.py

```diff
@@ -157,14 +160,21 @@ def dispatch(
                 enable_app_server=app_spec.enable_app_server,
                 flow_servers=app_spec.flow_servers,
                 image_spec=app_spec.image_spec,
+                cluster_id=cluster_id,
```

Comment:
I am not sure coupling app releases with a cluster is the right thing to do here. Apps releases are like new app versions that's available in the app store. The cluster coupling comes up when somebody tries to run this app version and hence should be an attribute of app instance (few lines below this)

Reply:
I am not even sure why we have `cluster-id` in the release spec at all

Reply:
From manual verification, the cluster didn't start on the selected cluster without passing this parameter to the release.

I think we need to circle back on our service design and omit this in the API if we don't need it. For now it's required to support this sadly.